include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,184 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+         * @param[out] values The values of the B-splines evaluated at coordinate x. It has to be a (1+degree) 1D mdspan.
+         * @param[in] x The coordinate where B-splines are evaluated.
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_basis(DSpan1D values, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates B-splines derivatives at a given coordinate
+         *
+         * The derivatives are computed for every B-splines at the given coordinate x. A spline approximation of a derivative at coordinate x is a linear combination of those B-splines derivatives weigthed with splines coefficients of the spline-transformed initial discrete function.
+         * @param[out] derivs The derivatives of the B-splines evaluated at coordinate x. It has to be a (1+degree) 1D mdspan.
+         * @param[in] x The coordinate where B-splines derivatives are evaluated.
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type
         eval_deriv(DSpan1D derivs, ddc::Coordinate<Tag> const& x) const;
 
+        /** @brief Evaluates B-splines values and n derivatives at a given coordinate
+         *
+         * The values and derivatives are computed for every B-splines at the given coordinate x.
+         * @param[out] derivs The values and n derivatives of the B-splines evaluated at coordinate x. It has to be a (1+degree)*(1+n) 2D mdspan.
+         * @param[in] x The coordinates where B-splines derivatives are evaluated.
+         * @param[in] n The number of derivatives to evaluate (in addition to the B-splines values themselves).
+         * @return TODO
+         */
         KOKKOS_INLINE_FUNCTION discrete_element_type eval_basis_and_n_derivs(
                 ddc::DSpan2D derivs,
                 ddc::Coordinate<Tag> const& x,
                 std::size_t n) const;
 
+        /** @brief Compute the integrals of the B-splines.
+         *
+         * @param[out] int_vals The values of the integrals. It has to be a (1+nbasis) 1D mdspan.
+         * @return The values of the integrals.
+         */
         template <class Layout, class MemorySpace2>
         KOKKOS_INLINE_FUNCTION ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2>
         integrals(
                 ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2> int_vals) const;
 
+        /** @brief Returns the coordinate of the knot corresponding to the given index for a B-spline.
+         *
+         * @param[in] knot_idx Integer identifying index of the knot.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
         }
```

Comment:
```suggestion
        /** @brief Returns the coordinate of the knot corresponding to the given index.
         *
         * @param[in] knot_idx Integer identifying index of the knot.
         * @return Coordinate of the knot.
         */
        KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
        {
            return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
        }
```

This knot is not related to any particular B-spline

Reply:
Following comments below I have realised that there is information missing here:
```suggestion
        /** @brief Returns the coordinate of the knot corresponding to the given index for a B-spline.
         *
         * Returns the coordinate of the knot corresponding to the given index for a B-spline. The domain
         * over which the B-splines are defined is comprised of ncells+1 knots however there are a total of
         * ncells+1+2*degree knots. The additional knots which control the shape of the B-splines near the
         * boundary are added before and after the break points. The knot index is therefore in the interval [-degree, ncells+degree]
         *
         * @param[in] knot_idx Integer identifying index of the knot.
         * @return Coordinate of the knot.
         */
        KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
        {
            return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
        }
```